test/TestHelpers.jl

```diff
@@ -198,6 +198,24 @@ end
     val
 end
 
+@inline function Base.deleteat!(A::OffsetArray, i::Int)
+    checkbounds(A, i)
```

Comment:
I still think making a separate PR to put all the `checkbounds` calls for `OffsetArray`s inside an `@boundscheck` would keep things more organized and consistent.